phoenix-core/src/main/antlr3/PhoenixSQL.g

```diff
@@ -726,8 +727,11 @@ upsert_column_refs returns [Pair<List<ColumnDef>,List<ColumnName>> ret]
 
 // Parse a full declare cursor expression structure.
 declare_cursor_node returns [DeclareCursorStatement ret]
-    :    DECLARE c=cursor_name CURSOR FOR s=select_node
-        {ret = factory.declareCursor(c, s); }
+@init{boolean isStatic = false;}
+    :   DECLARE c=cursor_name CURSOR
+        (STATIC {isStatic = true;})?
```

Comment:
here it can be replaced with, (static=STATIC)? and use factory.declareCursor(c,s static!=null)